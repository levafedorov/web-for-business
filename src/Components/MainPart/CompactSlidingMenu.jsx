import React, {Component} from 'react';
import './SlidingMenu.scss'
import slidingMenuHOC from './slidingMenuHOC';
import PropTypes from 'prop-types';


 class CompactSlidingMenu extends Component{
    
 
    render(){
       
  
        const {items, position} = this.props;

        return <nav className="main-menu compact" style={position}>
            <ul className="menu-list">
    { Object.keys(items).map(item => <li key={item}><a href={items[item]}>{item}</a></li>) }
            </ul>
            <button type='button' className="menu-down-btn">
              &darr;
            </button>
        </nav>
    }

}

CompactSlidingMenu.defaultProps = {
    items: {first:'first', second:'second', third:'third'},
  }
  
  
  CompactSlidingMenu.propTypes = {
      items: PropTypes.objectOf(PropTypes.string),
      position: PropTypes.object
  }
  



export const CompactSlidingMenuImpl = slidingMenuHOC(CompactSlidingMenu);
export const CompactSlidingMenuTest = CompactSlidingMenu;