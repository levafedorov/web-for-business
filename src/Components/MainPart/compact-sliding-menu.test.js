import React from 'react';
import {CompactSlidingMenuTest} from './CompactSlidingMenu';
import {shallow, mount} from 'enzyme';

const props = {items: {home: 'home', about: 'about', services: 'services'}}





describe('CompactSlidingMenuTest component', () => {
    const setUp = (props) => mount(<CompactSlidingMenuTest {...props}/>);
    let component;
    let instance;
    beforeEach(() => {
        component = setUp(props);
        instance = component.instance();
    });

    it("Should components prop's to be working", () => {
        component = setUp();
        expect(component).toMatchSnapshot();
        component = setUp(props);
        expect(component).toMatchSnapshot();
    });

//     describe('Component methods', () => {
      
            
// });

});
