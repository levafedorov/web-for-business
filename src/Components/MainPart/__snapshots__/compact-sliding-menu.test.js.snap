// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompactSlidingMenuTest component Should components prop's to be working 1`] = `
<CompactSlidingMenu
  items={
    Object {
      "first": "first",
      "second": "second",
      "third": "third",
    }
  }
>
  <nav
    className="main-menu compact"
  >
    <ul
      className="menu-list"
    >
      <li
        key="first"
      >
        <a
          href="first"
        >
          first
        </a>
      </li>
      <li
        key="second"
      >
        <a
          href="second"
        >
          second
        </a>
      </li>
      <li
        key="third"
      >
        <a
          href="third"
        >
          third
        </a>
      </li>
    </ul>
    <button
      className="menu-down-btn"
      type="button"
    >
      ↓
    </button>
  </nav>
</CompactSlidingMenu>
`;

exports[`CompactSlidingMenuTest component Should components prop's to be working 2`] = `
<CompactSlidingMenu
  items={
    Object {
      "about": "about",
      "home": "home",
      "services": "services",
    }
  }
>
  <nav
    className="main-menu compact"
  >
    <ul
      className="menu-list"
    >
      <li
        key="home"
      >
        <a
          href="home"
        >
          home
        </a>
      </li>
      <li
        key="about"
      >
        <a
          href="about"
        >
          about
        </a>
      </li>
      <li
        key="services"
      >
        <a
          href="services"
        >
          services
        </a>
      </li>
    </ul>
    <button
      className="menu-down-btn"
      type="button"
    >
      ↓
    </button>
  </nav>
</CompactSlidingMenu>
`;
